cpp/src/arrow/util/rle_encoding_internal.h

```diff
@@ -118,50 +365,56 @@ class RleDecoder {
   /// input with zeros. Since the encoding does not differentiate between
   /// input values and padding, Get() returns true even for these padding
   /// values.
-  template <typename T>
-  bool Get(T* val);
+  [[nodiscard]] bool Get(value_type* val);
```

Comment:
I'm curious, is this method actually used somewhere? It sounds unexpected to decode data one value at a time.